### PR TITLE
asymptote: fix build on aarch64-darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

ZHF: #199919

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2164,7 +2164,6 @@ with pkgs;
 
   asymptote = callPackage ../tools/graphics/asymptote {
     texLive = texlive.combine { inherit (texlive) scheme-small epsf cm-super texinfo media9 ocgx2; };
-    gsl = gsl_1;
```
Comment (Atemu) [inline pkgs/top-level/all-packages.nix:2167]: Why was this here and why isn't it required anymore? Would be good to have that info in the commit message.

**Review by Atemu — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 199258` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>asymptote</li>
  </ul>
</details>


```
error: builder for '/nix/store/600n8c85h74wwddp7l19ch015900gnp5-cups-2.2.6.drv' failed with exit code 1;
       last 10 log lines:
       > applying patch /nix/store/9yg7j3h4yrms07vgb1bcw0d5l2sn5b6f-CVE-2022-26691-fix-comment.patch
       > patching file scheduler/cert.c
       > updateAutotoolsGnuConfigScriptsPhase
       > Updating Autotools / GNU config script to a newer upstream version: ./config.sub
       > Updating Autotools / GNU config script to a newer upstream version: ./config.guess
       > configuring
       > configure flags: --disable-static --prefix=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6 --bindir=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6/bin --sbindir=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6/sbin --includedir=/nix/store/g208n34dssif0ph3ffcmpm4wp6lcykhi-cups-2.2.6-dev/include --oldincludedir=/nix/store/g208n34dssif0ph3ffcmpm4wp6lcykhi-cups-2.2.6-dev/include --mandir=/nix/store/1ldz692w6j0z98bdawda6syi11x7274d-cups-2.2.6-man/share/man --infodir=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6/share/info --docdir=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6/share/doc/cups --libdir=/nix/store/0jrzl658sb18zfyzk2kbry1ahhxgngzm-cups-2.2.6-lib/lib --libexecdir=/nix/store/0jrzl658sb18zfyzk2kbry1ahhxgngzm-cups-2.2.6-lib/libexec --localedir=/nix/store/0jrzl658sb18zfyzk2kbry1ahhxgngzm-cups-2.2.6-lib/share/locale --localstatedir=/var --sysconfdir=/etc --enable-raw-printing --enable-threads --enable-libusb --enable-ssl --enable-avahi --enable-libpaper --disable-launchd --build=aarch64-apple-darwin --host=aarch64-apple-darwin --prefix=/nix/store/0jrzl658sb18zfyzk2kbry1ahhxgngzm-cups-2.2.6-lib --datadir=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6/share --localedir=/nix/store/0jrzl658sb18zfyzk2kbry1ahhxgngzm-cups-2.2.6-lib/share/locale --with-systemd=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6/lib/systemd/system --with-bundledir=/nix/store/hbf85hgi8n1d2mp08ns23w6plxa7dnaf-cups-2.2.6
       > checking build system type... aarch64-apple-darwin
       > checking host system type... aarch64-apple-darwin
       > configure: error: Sorry, this version of CUPS requires macOS 10.8 or higher.
       For full logs, run 'nix log /nix/store/600n8c85h74wwddp7l19ch015900gnp5-cups-2.2.6.drv'.
error: 1 dependencies of derivation '/nix/store/s25h152cz9j2xvwdvvd9s9zvkgpn2rmw-ghostscript-with-X-9.56.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sl4wbmi3sk672phpkmivid7mr0lk3yd8-qtbase-5.15.7.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7jf00qwhdhlvjc6jbw1l5hmg1dkv1sdp-hook.drv' failed to build
error: 2 dependencies of derivation '/nix/store/lnfpalyigihfi29kww44l51fdvl10xl0-qtsvg-5.15.7.drv' failed to build
error: 3 dependencies of derivation '/nix/store/bh739v6b2nbbc2mqpryin680lwl83pf0-qtdeclarative-5.15.7.drv' failed to build
error: 3 dependencies of derivation '/nix/store/g4qjnqprl7rkmfaz9nq2n25k41q8v0ya-qtwebchannel-5.15.7.drv' failed to build
error: 5 dependencies of derivation '/nix/store/nvbq2pbhp0zm2rx5ws6z8ls6lr662zqq-python3.10-PyQt5-5.15.7.drv' failed to build
error: builder for '/nix/store/8mifijl9crxhnnjr9lkdg3229yhnq34v-mesa-22.2.2.drv' failed with exit code 1;
       last 10 log lines:
       > Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/meson-private/glesv2.pc to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/pkgconfig
       > Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/meson-private/dri.pc to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/pkgconfig
       > Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/meson-private/gl.pc to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/pkgconfig
       > Installing symlink pointing to libglapi.0.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libglapi.dylib
       > Installing symlink pointing to libGLESv1_CM.1.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libGLESv1_CM.dylib
       > Installing symlink pointing to libGLESv2.2.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libGLESv2.dylib
       > Installing symlink pointing to libGL.1.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libGL.dylib
       > Running custom install script '/nix/store/14sv2h3nag638062j87lg3ic4z8ipgis-python3-3.10.8/bin/python3.10 /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/bin/install_megadrivers.py /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/src/gallium/targets/dri/libgallium_dri.so /nix/store/7pmab9bri6pc4y5phkmfdqlwavmc01jx-mesa-22.2.2-drivers/lib/dri swrast_dri.so'
       > mv: missing file operand
       > Try 'mv --help' for more information.
       For full logs, run 'nix log /nix/store/8mifijl9crxhnnjr9lkdg3229yhnq34v-mesa-22.2.2.drv'.
error: 1 dependencies of derivation '/nix/store/qjyaxm2k2s00y6d8sa4j0za93bqdbg05-libGL-1.5.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/81a9f37j6dxb4q56jh9gxjja2s4pjd9a-libdevil-1.7.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/k6s3z69c58zh531v7h4bzw827mbjsxgw-graphviz-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/m7vlahfa0rz57z60c1z3nn007kccqbf6-libjxl-0.6.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kc7mrwyj35832hg9hkinpzkrkalpbqn9-imagemagick-7.1.0-51.drv' failed to build
error: 3 dependencies of derivation '/nix/store/q0q41s92inllaksp2g95gxvzxw5r05a7-asymptote-2.83.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ns6wsxpckpyam0xqb6a3qwsnlad6x93x-review-shell.drv' failed to build
```

**Review by Atemu — COMMENTED** [review]
Dependencies are broken on master.

Result of `nixpkgs-review pr 199258` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>asymptote</li>
  </ul>
</details>

I've let it draw something but Preview wasn't able to convert the postscript to a PDF.

**Comment by wegank** [comment]
I tested the package against [this](https://artofproblemsolving.com/wiki/index.php/Asymptote:_Macintosh).

**Comment by wegank** [comment]
@ofborg build asymptote

**Review by Atemu — APPROVED** [review]
Result of `nixpkgs-review pr 199258` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>asymptote</li>
  </ul>
</details>

**Pull request merged**